dev/com.ibm.ws.security.mp.jwt.1.2_fat/fat/src/com/ibm/ws/security/mp/jwt12/fat/sharedTests/MPJwt12MPConfigTests.java

```diff
@@ -519,58 +650,191 @@ public Expectations setBadAudiencesExpectations(LibertyServer server) throws Exc
+        Expectations expectations = setAllBadEncryptExpectations(server, extraMsgs);
+        expectations.addExpectation(new ServerMessageExpectation(server, "The key must not be null", "Messagelog did not contain an exception indicating that the keyManagementKeyAlias was missing."));
+        return expectations;
+
+    }
+
+    /**
+     * Sets expectations to check when the keyManagementKeyAlias is not set
+     *
+     * @param server - server whose logs will be searched
+     * @param extraMsgs - the tai drives the code down different paths depending on if it finds config info in server.xml - if it finds config settings, we'll get 2 extra messages.
+     * @return Expectations - built expectations
+     * @throws Exception
+     */
+    public Expectations setEncryptPlainTextKeyExpectations(LibertyServer server, boolean extraMsgs) throws Exception {
+        Expectations expectations = setAllBadEncryptExpectations(server, extraMsgs);
+        expectations.addExpectation(new ServerMessageExpectation(server, MpJwtMessageConstants.CWWKS6062E_PLAINTEXT_KEY, "Messagelog did not contain an exception indicating that the mp.jwt.decrypt.key.location contained a plaintext key."));
+        return expectations;
+
+    }
+
+    /**
+     * Sets expectations to check when the keyManagementKeyAlias is not set
+     *
+     * @param server - server whose logs will be searched
+     * @param extraMsgs - the tai drives the code down different paths depending on if it finds config info in server.xml - if it finds config settings, we'll get 2 extra messages.
+     * @return Expectations - built expectations
+     * @throws Exception
+     */
+    public Expectations setEncryptMismatchExpectations(LibertyServer server, boolean extraMsgs) throws Exception {
+        Expectations expectations = setAllBadEncryptExpectations(server, extraMsgs);
+        expectations.addExpectation(new ServerMessageExpectation(server, "javax.crypto.AEADBadTagException", "Messagelog did not contain an exception indicating a tag mismatch."));
+        return expectations;
+
+    }
+
+    /**
+     * Sets expectations to check when the keyManagementKeyAlias is not set
+     *
+     * @param server - server whose logs will be searched
+     * @param extraMsgs - the tai drives the code down different paths depending on if it finds config info in server.xml - if it finds config settings, we'll get 2 extra messages.
+     * @return Expectations - built expectations
+     * @throws Exception
+     */
+    public Expectations setEncryptMismatchKeyTypeExpectations(LibertyServer server, boolean extraMsgs) throws Exception {
+        Expectations expectations = setAllBadEncryptExpectations(server, extraMsgs);
+        expectations.addExpectation(new ServerMessageExpectation(server, "java.lang.ClassCastException", "Messagelog did not contain an exception indicating a classcast exception due to the key type."));
```

Comment:
I might be interested in double checking this one. Do you know whether the `ClassCastException` is coming from our code or from jose4j? I wonder if there might be something cleaner we could do, depending on what this actually ends up looking like.